### PR TITLE
feat(casing): Add upperCase and lowerCase

### LINKED ISSUE #1510: _.downcase and _.upcase requested.
For some reason, I'm not able to pass in `String.prototype.toLowerCase.call` as a function object to `.then` in protractor. Not sure it'd be a major priority for you, but I'd appreciate it if I could pass in `_.downcase` to `.then` and not have to write it all out everytime.

```js
return myElement.getText().then(function (text) {
    return text.toLowerCase();
});
```

versus

```js
return myElement.getText().then(_.downcase);
```

I feel this would compliment `camelCase`, `snakeCase`, and other casing helpers.

### PR DESCRIPTION
**Pull request by Droogans**
Closes #1510

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -1867,6 +1867,78 @@
...
+      }
+    });
+  }());
+
+  /*--------------------------------------------------------------------------*/
+
+  QUnit.module('lodash.upperCase');
+
+  (function() {
+    QUnit.test('should lower case the entire string', function(assert) {
```
Comment (Droogans) [inline test/test.js]: I'll fix this.

**Comment by jdalton** [comment]
Have you signed [our CLA](http://dojofoundation.org/about/claForm)?

**Comment by Droogans** [comment]
Just did.

**Comment by jdalton** [comment]
Rock!

**Pull request merged**